cli/lock.ml

```diff
@@ -311,26 +312,27 @@ let local_packages ~versions repo =
 let run (`Root root) (`Recurse_opam recurse) (`Build_only build_only)
     (`Allow_jbuilder allow_jbuilder) (`Ocaml_version ocaml_version)
     (`Target_packages specified_packages) (`Lockfile explicit_lockfile) () =
-  let open Rresult.R.Infix in
-  local_packages ~versions:specified_packages root >>= fun local_packages ->
-  target_packages ~local_packages ~recurse
-    ~explicitly_specified:specified_packages root
-  >>= fun target_packages ->
-  check_target_packages target_packages >>= fun () ->
-  local_paths_to_opam_map local_packages >>= fun opam_files ->
-  lockfile_path ~explicit_lockfile ~target_packages root
-  >>= fun lockfile_path ->
-  calculate_opam ~build_only ~allow_jbuilder ~ocaml_version
-    ~local_opam_files:opam_files ~target_packages
-  >>= fun package_summaries ->
+  let open Result.O in
+  let* local_packages = local_packages ~versions:specified_packages root in
+  let* target_packages =
+    target_packages ~local_packages ~recurse
+      ~explicitly_specified:specified_packages root
+  in
+  let* () = check_target_packages target_packages in
```

Comment:
Agreed! I am curious about the omission of `;*` and `match*` in the feature. It feels… half-baked somewhat? Well, maybe in the future. Or maybe this could be eventually subsumed by effects.